helmfile/charts/networkpolicy-common/templates/coredns/allow-coredns.yaml

```diff
@@ -0,0 +1,49 @@
+{{ if .Values.coredns.enabled }}
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: allow-coredns
+spec:
+  podSelector:
+    matchLabels:
+      k8s-app: kube-dns
+  policyTypes:
+    - Egress
+    - Ingress
+  ingress:
+    - ports:
+      - port: 53
+        protocol: UDP
+      - port: 53
+        protocol: TCP
+    - from:
+        - podSelector:
+            matchLabels:
+              app: prometheus
+          namespaceSelector:
+            matchLabels:
+              kubernetes.io/metadata.name: monitoring
+      ports:
+        - port: 9153
+  egress:
+    - to:
+      {{- range $IP := .Values.global.apiserver.ips }}
+      - ipBlock:
+            cidr: {{ $IP }}
+        {{- end }}
+      ports:
+        - port: {{ .Values.global.apiserver.port }}
+    - to:
+      {{- range $IP := .Values.coredns.externalDns.ips }}
+      - ipBlock:
+          cidr: {{ $IP }}
+      {{- end }}
+      - ipBlock:
+          cidr: 10.233.0.3/32
```

Comment:
Is coredns talking to itself?

Reply:
Yes
I has 2 pods that are talking to each-other

Reply:
If so, could you insteadchange this to use the podselector instead?

```
- from:
    podSelector:
      matchLabels:
        k8s-app: kube-dns
```

I think that would be easier to understand

Reply:
even if it communicates via the service ip?

Reply:
Yes, the sevice IP doesn't matter since it will use the pod IP in the end.

Reply:
I have tried to remove the service ip and replace it with the pod selector but aparently the node-local-dns binds to the service ip and then when I use podSelector instead of the service ip, the packages to that node-local-dns are being dropped.
Both me and @aarnq could find an alternative solution to this
So I will keep the service ip instead of podSelector